Lib/random.py

```diff
@@ -464,11 +464,13 @@ def choices(self, population, weights=None, *, cum_weights=None, k=1):
         """
         random = self.random
         n = len(population)
+        if n == 0:
+            raise IndexError('Cannot choose from an empty population')
         if cum_weights is None:
             if weights is None:
                 floor = _floor
                 n += 0.0    # convert to float for a small speed improvement
-                return [population[floor(random() * n)] for i in _repeat(None, k)]
+                return [population[floor(random() * n)] for _ in _repeat(None, k)]
```

Comment:
Please avoid stylistic only changes.

Reply:
Hi nineteendo, thanks for the comment.

This change is included as miscellaneous updates bundled with other important changes. As you may notice, there is another similar change [here](https://github.com/python/cpython/pull/118962/files#diff-851cf2d8d99b6d8dee26acc31319431dce51dbe26e3e56494d9b4c45edd8e11fR498), which can avoid confusion because we added an index variable `i` [before](https://github.com/python/cpython/pull/118962/files#diff-851cf2d8d99b6d8dee26acc31319431dce51dbe26e3e56494d9b4c45edd8e11fR490).